Fix table not updating on second CSV upload

interianow opened the pull request:
## Description
Doing a second submit on the single form page was failing to recalculate usageData because a flag to perform the recalc was not getting reset after the first submit. The flag was created to a fix a previous bug where checkbox was not getting toggled because stale submitted actionData was overriding changes made locally.

I did away with the flag and created a useEffect that watches when actionData changes. If we kept the flag and reset it, we would bring back the original checkbox bug. Now, when a submit happens the useEffect will fire and recalc usageData. If user makes only local changes, then actionData does not change and the usageData data is not reset with stale data.

## Changes 
1. Moved logic for maintaining `usageData` into `useRulesEngine`. This includes recalculating and updating checkbox status on clicking table row.
3. Refactored `<EnergyUseHistoryChart />` to stop recalculating usageData and instead let `<SubmitAnalysis />` handle it
   - Previous implementation  made it difficult to track where changes to usageData was happening and which piece of data was being read. All of the data lived in `<SubmitAnalysis />` any ways so it made more sense to make `<EnergyUseHistoryChart />` a dumb component that only tells you which row was clicked.
4. Renamed `parsedLastResult` to `usageDataMap` and `lastResult` to `actionData` to give more meaning. 
    - `lastResult` was always getting set to `actionData` with no modifications which made reading the code especially difficult to understand
6. Fixed broken test in `energy-analysis.test.ts`. Test was broken because setup code failed to find an input after a previous refactor was made.
7. Split single long test in `energy-analysis.test.ts` to multiple smaller test blocks to make it easier to manage different scenarios.  
8. Added `heat-stack/tests/fixtures/csv/natural-gas-eversource.csv` to test submitting two csvs back to back
9. thad and ben fixed issue where form inputs were reset when submitting csv multiple times

## Side notes
- There is an existing bug where if you submit a second csv, some of the inputs on the form get cleared. This happens in prod and in this PR. For somereason, conform wants to reset the form fields on the second upload, not sure why. But since that is an existing issue independent of the work done here, I thought it would be best to submit this PR anyways.

- Some files show more diffs than was actually done. For example, `CustomFileUpload.tsx` I added a single `data-testid` attribute, but a large chunk of the file was marked as modified due to prettier formatting. It might be a good idea to add a pre-commit hook to take care of linting and formatting

thadk reviewed (approved):
This looks great, though we didn't check over everything! We stacked another commit on it to fix the disappearing conform form field values you reported: [interianow-fix/multi-submit-bug](https://github.com/codeforboston/home-energy-analysis-tool/tree/interianow-fix/multi-submit-bug)
https://github.com/codeforboston/home-energy-analysis-tool/commit/e0dfd706c7700e7a123ef9db26304339b14cb39b

We had a few comments in the review, but I think we mostly talked them through more deeply during the hack night.

interianow commented:
Made the changes discussed during the hackathon. I moved use-usage-data into use-rules-engine so there is only one hook now. I also added the changes to fix the disappearing form input values.

interianow commented:
I am going to close this PR and resubmit. I was running into issues trying to rebase/merge so I am just going to create a new PR off latest main.
